src/nplinker/pairedomics/podp_antismash_downloader.py

```diff
@@ -176,33 +245,44 @@ def _resolve_jgi_accession(jgi_id):
     return _resolve_genbank_accession(link.text)
 
 
-def _resolve_genome_id_data(genome_id_data):
+def _resolve_refseq_id(genome_id_data: Dict[str, str]) -> str | None:
+    """Get the RefSeq ID to which the genome accession is linked.
+    Check https://pairedomicsdata.bioinformatics.nl/schema.json.
+
+    Args:
+        genome_id_data(dict): dictionary containing information
+        for each genome record present.
+
+    Returns:
+        str | None: ID for the accession genome, if present, otherwise None. 
+        """
     if 'RefSeq_accession' in genome_id_data:
         # best case, can use this directly
         return genome_id_data['RefSeq_accession']
-    elif 'GenBank_accession' in genome_id_data:
+    if 'GenBank_accession' in genome_id_data:
         # resolve via NCBI
         return _resolve_genbank_accession(genome_id_data['GenBank_accession'])
-    elif 'JGI_Genome_ID' in genome_id_data:
+    if 'JGI_Genome_ID' in genome_id_data:
         # resolve via JGI => NCBI
         return _resolve_jgi_accession(genome_id_data['JGI_Genome_ID'])
 
     logger.warning(f'Unable to resolve genome_ID: {genome_id_data}')
     return None
 
 
-def _get_antismash_db_page(genome_obj):
+def _get_antismash_db_page(genome_obj: GenomeStatus) -> NavigableString | None:
```

Comment:
```suggestion
def _get_antismash_filename(genome_obj: GenomeStatus) -> str | None:
```
This method actually tries to get antismash filename by requesting database website.
Please add docstrings for it and make sure the returned type is a string.

Reply:
Same as above, if I leave it as `str | None`, I get "Returning Any from function declared to return "Optional[str]" from mypy

Reply:
it's probably caused by the situation that the type of returned value cannot be identified. Maybe converting the returned value to intended type would solve the issue, .e.g `return str(link['href'])`